File: src/rust/src/controllers/users_controller.rs
use crate::services;
use rocket::{http::Status, serde::json::Json};
use crate::models::user::User;

#[get("/<id>")]
pub async fn get_user(id: u32) -> Result<Json<User>, Status> {
    let user = services::users_service::get_user(id).await;
    match user {
        Ok(user) => Ok(Json(user)),
        Err(_) => Err(Status::NotFound),
    }
}

File: src/go/controllers/usersController.go
package controllers

import (
	"app/services"
	"github.com/gin-gonic/gin"
	"net/http"
	"strconv"
)

func GetUser(c *gin.Context) {

	id, err := strconv.Atoi(c.Param("id"))
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"message": "invalid path parameter"})
	}

	user, err := services.GetUser(id)
	if err != nil {
		c.JSON(http.StatusNotFound, gin.H{"message": "user not found"})
	}
	c.JSON(http.StatusOK, user)
}
